### ISSUE TITLE
[Feature request] @mention support?

### ISSUE THREAD

**Issue by dcsan** [Chatie/wechaty#153]
In a group it is so convenient to message the bot using @botname

I was looking at the [message API](https://github.com/wechaty/wechaty/wiki/API#message-class) and trying to see how @mentions work

but looking at output from ` console.log(inspect(m))` . I don't see any special data that refers to the mentioned users. 
Their names are included in the text of the message. If you change your name, then the mention will use the updated name (not the wechat ID)

So I guess the only way is to use message and match against the `Room.memberList()`
It just seems surprising this is left encoded in the message text and not as a userId in the data somewhere...

I guess that the memberList requires room.ready() and maybe not available all the time?
So we could provide this as an async API helper on the message object?

**Comment by zixia** [comment]
Thanks for the great feature suggestion!

I agree that if we have a `room-mention` / `mention` event for the room when we got mentioned will be very convenience.

How about we design it like this: 

```ts
// Global Event
Wechaty.instance()
.on('room-mention', (room, message) => {
  // here we got message mentioned us
})

// or Room Event

Room.find({ name: 'test' })
.then(room => {
  room.on('mention', message => {
    // here we got message mentioned us
  })
})
```

**Comment by lijiarui** [comment]
It may fix as it can on #362

**Comment by zixia** [comment]
I'll close this issue because we have `message.mentioned()` now.
